python/nano/src/bigdl/nano/pytorch/amp/bfloat16.py

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import contextlib
+import io
+from logging import info, warning
+
+from pytorch_lightning import LightningModule
+import torch
+import subprocess
+import os
+from bigdl.nano.utils.log4Error import invalidOperationError, invalidInputError
+from bigdl.nano.pytorch.utils import TORCH_VERSION_LESS_1_10, TORCH_VERSION_LESS_1_12,\
+    LIGHTNING_VERSION_LESS_1_6
+
+invalidInputError(
+    not TORCH_VERSION_LESS_1_10 and not TORCH_VERSION_LESS_1_12 and not LIGHTNING_VERSION_LESS_1_6,
```

Comment:
`not TORCH_VERSION_LESS_1_10` is redundant right

Reply:
Not really. I just modified the error to give messages when

- version<1.10, cannot convert model to bf16 type
- 1.10 < version < 1.12, users can obtain bf16 model, but no acceleration, so `model(x)` is not allowed.
- version < 1.12, when bf16 instructions are available, `model(x)` is allowed